### ISSUE TITLE
failed to run make build

### ISSUE THREAD

**Issue by newTechAccount** [RediSearch/RediSearch#2437]
centos 7.x,gcc 4.8.5 -->10.x
`Building /home/tools/RediSearch/bin/linux-x64-release/search/redisearch.so ...
[  0%] Performing build step for 'VectorSimilarity'
[  4%] Building CXX object VecSim/spaces/CMakeFiles/VectorSimilaritySpaces.dir/space_aux.cpp.o
/home/tools/RediSearch/deps/VectorSimilarity/src/VecSim/spaces/space_aux.cpp:1:0: error: bad value (skylake-avx512) for -march= switch
 #include "space_aux.h"
 ^
make[6]: *** [VecSim/spaces/CMakeFiles/VectorSimilaritySpaces.dir/build.make:76: VecSim/spaces/CMakeFiles/VectorSimilaritySpaces.dir/space_aux.cpp.o] Error 1
make[5]: *** [CMakeFiles/Makefile2:126: VecSim/spaces/CMakeFiles/VectorSimilaritySpaces.dir/all] Error 2
make[4]: *** [Makefile:136: all] Error 2
make[3]: *** [CMakeFiles/VectorSimilarity.dir/build.make:86: ../VectorSimilarity/src/VectorSimilarity-stamp/VectorSimilarity-build] Error 2
make[2]: *** [CMakeFiles/Makefile2:364: CMakeFiles/VectorSimilarity.dir/all] Error 2
make[1]: *** [Makefile:146: all] Error 2
make: *** [Makefile:366: /home/tools/RediSearch/bin/linux-x64-release/search/redisearch.so] Error 2`

**Comment by rafie** [comment]
@newTechAccount Thank you for not using capital letters.
Can you please elaborate more?
Did you run `make setup`?

**Comment by newTechAccount** [comment]
> @newTechAccount Thank you for not using capital letters. Can you please elaborate more? Did you run `make setup`?
yes,i did ran 'sudo make setup' without any error.
after make up,run 'make build',then i got 

> [root@VM-4-3-centos RediSearch]# make build
# search root: /home/tools/RediSearch
# search binroot: /home/tools/RediSearch/bin/linux-x64-release
RediSearch/VecSim VECSIM_MARCH: skylake-avx512
RediSearch/VecSim dir: /home/tools/RediSearch/deps/VectorSimilarity/src
RediSearch/VecSim bindir: /home/tools/RediSearch/bin/linux-x64-release/VectorSimilarity
-- Configuring done
-- Generating done
-- Build files have been written to: /home/tools/RediSearch/bin/linux-x64-release/search
Building /home/tools/RediSearch/bin/linux-x64-release/search/redisearch.so ...
[  0%] Performing build step for 'VectorSimilarity'
[  4%] Building CXX object VecSim/spaces/CMakeFiles/VectorSimilaritySpaces.dir/space_aux.cpp.o
/home/tools/RediSearch/deps/VectorSimilarity/src/VecSim/spaces/space_aux.cpp:1:0: error: bad value (skylake-avx512) for -march= switch
 #include "space_aux.h"
 ^
make[6]: *** [VecSim/spaces/CMakeFiles/VectorSimilaritySpaces.dir/build.make:76: VecSim/spaces/CMakeFiles/VectorSimilaritySpaces.dir/space_aux.cpp.o] Error 1
make[5]: *** [CMakeFiles/Makefile2:126: VecSim/spaces/CMakeFiles/VectorSimilaritySpaces.dir/all] Error 2
make[4]: *** [Makefile:136: all] Error 2
make[3]: *** [CMakeFiles/VectorSimilarity.dir/build.make:86: ../VectorSimilarity/src/VectorSimilarity-stamp/VectorSimilarity-build] Error 2
make[2]: *** [CMakeFiles/Makefile2:364: CMakeFiles/VectorSimilarity.dir/all] Error 2
make[1]: *** [Makefile:146: all] Error 2
make: *** [Makefile:366: /home/tools/RediSearch/bin/linux-x64-release/search/redisearch.so] Error 2

**Comment by rafie** [comment]
Indeed, our experience with gcc 10 on CentOS 7 (via `readies/bin/getgcc --modern` delivering `centos-release-scl/devtoolset-10`), it does not support `-march x86-64-v4` but only `skylake-avx512` (which is our RediSearch build default - see `Makefile`).
Other Linux distributions (e.g. Ubuntu Focal) with similar gcc versions do support `-march x86-64-v4`, and I assume it's a matter of toolchain build configuration.
At the moment, we do not build our own toolchain for CentOS 7, but this may change in the future is there will be demand for it. This is of course outside the scope of RediSearch, and everyone may take a shot at building their own toolchain.
